_includes/cockroachcloud/cmek-rotation-types.md

```diff
@@ -0,0 +1,5 @@
+You can rotate a CMEK key within your KMS platform or within {{ site.data.products.dedicated }}.
+
+- When you rotate a CMEK key within your KMS platform, a new version of the key is created with new key material, and this becomes the active key version. In both GCP KMS and AWS KMS, encryption operations automatically use the active key version, while decryption operations automatically use the key version that was used to encrypt the data. For this reason, {{ site.data.products.dedicated }} does not need any awareness of rotation operations within your KMS platform. However, {{ site.data.products.db }} will not re-encrypt the store key using the rotated CMEK key until you also perform a rotation using the {{ site.data.products.db }} API.
```

Comment:
@adwittumuluri - This whole description looks ok to you? I'm in two mind about it - it's useful, but at the same time could be confusing for someone. Should we just have something like:

- If you configure CMEK key auto-rotation in your KMS platform, then your cloud provider changes the underlying key material and creates a new version of the key periodically. Else, you can manually create a new version of the existing CMEK key in your KMS platform (only possible in GCP, not in AWS). In both cases, {{ site.data.products.db }} will not re-encrypt the store key using the new version of the existing CMEK key until you perform a rotation with the {{ site.data.products.db }} API, while using the existing CMEK key reference. Do not disable or delete the previous version of the existing CMEK key until rotation process is complete.
- You can also create a new CMEK key altogether instead of rotating versions of an existing CMEK key. In that case, you supply key URI for the new CMEK for each region you want to update, and {{ site.data.products.dedicated }} begins using the new key to protect the store key. Do not disable or delete your previous CMEK key until rotation process is complete.

Reply:
I think Abhinav's bullets here make it a bit more clear what the customer should and shouldn't *do* when rotating versions (vs providing info). "[CC] does not need any awareness of rotation operations within your KMS platform" may not be information that the customer cares about, as what they have to _do_ is given explicitly. The difference is very slim though.